src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java

```diff
@@ -162,34 +162,47 @@ public void configure() {
                 .id("unzip-file")
                 .choice()
                     .when(isZippedFile("zip"))
-                        .split(new ZipSplitter())
+                        .split(new ZipSplitter()).aggregationStrategy(this::calculateICSAggregated)
                         .streaming()
                         .to("direct:calculate")
                     .endChoice()
                     .when(isZippedFile("tar.gz"))
                         .unmarshal().gzip()
-                        .split(new TarSplitter())
-                        .streaming()
-                        .to("direct:calculate")
+                        .split(new TarSplitter()).aggregationStrategy(this::calculateICSAggregated)
+                            .streaming()
+                            .to("direct:calculate")
+                        .end()
                     .endChoice()
                     .otherwise()
                         .to("direct:calculate")
-                .end();
+                        .process(exchange -> exchange.getIn().getHeaders().put("uploadformdata", exchange.getIn().getBody(UploadFormInputDataModel.class)))
+                    .end()
+                .end()
+                // at this time it's safe to generate the WSR
+                .to("direct:send-costsavings")
+                .to("direct:calculate-workloadsummaryreportmodel");
 
         from("direct:calculate")
                 .id("calculate")
                 .convertBodyTo(String.class)
-                .multicast()
+                .multicast().aggregationStrategy(new org.apache.camel.processor.aggregate.GroupedBodyAggregationStrategy())
                     .to("direct:calculate-costsavings", "direct:calculate-vmworkloadinventory", "direct:flags-shared-disks")
+                .setBody(e -> e.getIn().getBody(List.class).get(0))
                 .end();
 
-        from("direct:calculate-costsavings")
-                .id("calculate-costsavings")
-                .transform().method("calculator", "calculate(${body}, ${header.MA_metadata})")
+
+        from("direct:send-costsavings").id("send-costsavings")
+                .log("SEND-COSTSAVINGS")
```

Comment:
Could you add the `LoggingLevel.DEBUG` param please?

Reply:
In fact I will remove this LOG as it was only for testing purposes, with Camel Tracing is enough.